### PR TITLE
Adding connection to a Xilinx license server.

### PR DESCRIPTION
**Pull request by mithro**
Enables generation of database for larger Kintex parts.

Signed-off-by: Tim 'mithro' Ansell <tansell@google.com>

### CODE REVIEW

**Comment by mithro** [comment]
Was finally able to make this work!
```
Preparing kintex7 files
============================
Part xc7k160tfbg484-1 has an unsupported device xc7k160t
Part xc7k160tfbg484-2 has an unsupported device xc7k160t
Part xc7k160tfbg484-2L has an unsupported device xc7k160t
Part xc7k160tfbg484-3 has an unsupported device xc7k160t
Part xc7k160tfbg676-1 has an unsupported device xc7k160t
Part xc7k160tfbg676-2 has an unsupported device xc7k160t
Part xc7k160tfbg676-2L has an unsupported device xc7k160t
Part xc7k160tfbg676-3 has an unsupported device xc7k160t
Part xc7k160tfbv484-1 has an unsupported device xc7k160t
Part xc7k160tfbv484-2 has an unsupported device xc7k160t
Part xc7k160tfbv484-2L has an unsupported device xc7k160t
Part xc7k160tfbv484-3 has an unsupported device xc7k160t
Part xc7k160tfbv676-1 has an unsupported device xc7k160t
Part xc7k160tfbv676-2 has an unsupported device xc7k160t
Part xc7k160tfbv676-2L has an unsupported device xc7k160t
Part xc7k160tfbv676-3 has an unsupported device xc7k160t
Part xc7k160tffg676-1 has an unsupported device xc7k160t
Part xc7k160tffg676-2 has an unsupported device xc7k160t
Part xc7k160tffg676-2L has an unsupported device xc7k160t
Part xc7k160tffg676-3 has an unsupported device xc7k160t
Part xc7k160tffv676-1 has an unsupported device xc7k160t
Part xc7k160tffv676-2 has an unsupported device xc7k160t
Part xc7k160tffv676-2L has an unsupported device xc7k160t
Part xc7k160tffv676-3 has an unsupported device xc7k160t
Part xc7k160tifbv676-2L has an unsupported device xc7k160ti
Part xc7k160tifbg484-2L has an unsupported device xc7k160ti
Part xc7k160tifbg676-2L has an unsupported device xc7k160ti
Part xc7k160tifbv484-2L has an unsupported device xc7k160ti
Part xc7k160tiffv676-2L has an unsupported device xc7k160ti
Part xc7k160tiffg676-2L has an unsupported device xc7k160ti
Part xc7k325tiffv676-2L has an unsupported device xc7k325ti
Part xc7k325tiffv900-2L has an unsupported device xc7k325ti
Part xc7k325tifbg676-2L has an unsupported device xc7k325ti
Part xc7k325tifbg900-2L has an unsupported device xc7k325ti
Part xc7k325tifbv676-2L has an unsupported device xc7k325ti
Part xc7k325tifbv900-2L has an unsupported device xc7k325ti
Part xc7k325tiffg900-2L has an unsupported device xc7k325ti
Part xc7k325tiffg676-2L has an unsupported device xc7k325ti
Part xc7k325tffv900-1 has an unsupported device xc7k325t
Part xc7k325tffv900-2 has an unsupported device xc7k325t
Part xc7k325tffv900-2L has an unsupported device xc7k325t
Part xc7k325tffv900-3 has an unsupported device xc7k325t
Part xc7k325tffv676-1 has an unsupported device xc7k325t
Part xc7k325tffv676-2 has an unsupported device xc7k325t
Part xc7k325tffv676-2L has an unsupported device xc7k325t
Part xc7k325tffv676-3 has an unsupported device xc7k325t
Part xc7k325tffg900-1 has an unsupported device xc7k325t
Part xc7k325tffg900-2 has an unsupported device xc7k325t
Part xc7k325tffg900-2L has an unsupported device xc7k325t
Part xc7k325tffg900-3 has an unsupported device xc7k325t
Part xc7k325tffg676-1 has an unsupported device xc7k325t
Part xc7k325tffg676-2 has an unsupported device xc7k325t
Part xc7k325tffg676-2L has an unsupported device xc7k325t
Part xc7k325tffg676-3 has an unsupported device xc7k325t
Part xc7k325tfbv900-1 has an unsupported device xc7k325t
Part xc7k325tfbv900-2 has an unsupported device xc7k325t
Part xc7k325tfbv900-2L has an unsupported device xc7k325t
Part xc7k325tfbv900-3 has an unsupported device xc7k325t
Part xc7k325tfbv676-1 has an unsupported device xc7k325t
Part xc7k325tfbv676-2 has an unsupported device xc7k325t
Part xc7k325tfbv676-2L has an unsupported device xc7k325t
Part xc7k325tfbv676-3 has an unsupported device xc7k325t
Part xc7k325tfbg900-1 has an unsupported device xc7k325t
Part xc7k325tfbg900-2 has an unsupported device xc7k325t
Part xc7k325tfbg900-2L has an unsupported device xc7k325t
Part xc7k325tfbg900-3 has an unsupported device xc7k325t
Part xc7k325tfbg676-1 has an unsupported device xc7k325t
Part xc7k325tfbg676-2 has an unsupported device xc7k325t
Part xc7k325tfbg676-2L has an unsupported device xc7k325t
Part xc7k325tfbg676-3 has an unsupported device xc7k325t
Part xc7k355tiffg901-2L has an unsupported device xc7k355ti
Part xc7k355tiffv901-2L has an unsupported device xc7k355ti
Part xc7k355tffg901-1 has an unsupported device xc7k355t
Part xc7k355tffg901-2 has an unsupported device xc7k355t
Part xc7k355tffg901-2L has an unsupported device xc7k355t
Part xc7k355tffg901-3 has an unsupported device xc7k355t
Part xc7k355tffv901-1 has an unsupported device xc7k355t
Part xc7k355tffv901-2 has an unsupported device xc7k355t
Part xc7k355tffv901-2L has an unsupported device xc7k355t
Part xc7k355tffv901-3 has an unsupported device xc7k355t
Part xc7k410tfbg676-1 has an unsupported device xc7k410t
Part xc7k410tfbg676-2 has an unsupported device xc7k410t
Part xc7k410tfbg676-2L has an unsupported device xc7k410t
Part xc7k410tfbg676-3 has an unsupported device xc7k410t
Part xc7k410tfbg900-1 has an unsupported device xc7k410t
Part xc7k410tfbg900-2 has an unsupported device xc7k410t
Part xc7k410tfbg900-2L has an unsupported device xc7k410t
Part xc7k410tfbg900-3 has an unsupported device xc7k410t
Part xc7k410tfbv676-1 has an unsupported device xc7k410t
Part xc7k410tfbv676-2 has an unsupported device xc7k410t
Part xc7k410tfbv676-2L has an unsupported device xc7k410t
Part xc7k410tfbv676-3 has an unsupported device xc7k410t
Part xc7k410tfbv900-1 has an unsupported device xc7k410t
Part xc7k410tfbv900-2 has an unsupported device xc7k410t
Part xc7k410tfbv900-2L has an unsupported device xc7k410t
Part xc7k410tfbv900-3 has an unsupported device xc7k410t
Part xc7k410tffg676-1 has an unsupported device xc7k410t
Part xc7k410tffg676-2 has an unsupported device xc7k410t
Part xc7k410tffg676-2L has an unsupported device xc7k410t
Part xc7k410tffg676-3 has an unsupported device xc7k410t
Part xc7k410tffg900-1 has an unsupported device xc7k410t
Part xc7k410tffg900-2 has an unsupported device xc7k410t
Part xc7k410tffg900-2L has an unsupported device xc7k410t
Part xc7k410tffg900-3 has an unsupported device xc7k410t
Part xc7k410tffv676-1 has an unsupported device xc7k410t
Part xc7k410tffv676-2 has an unsupported device xc7k410t
Part xc7k410tffv676-2L has an unsupported device xc7k410t
Part xc7k410tffv676-3 has an unsupported device xc7k410t
Part xc7k410tffv900-1 has an unsupported device xc7k410t
Part xc7k410tffv900-2 has an unsupported device xc7k410t
Part xc7k410tffv900-2L has an unsupported device xc7k410t
Part xc7k410tffv900-3 has an unsupported device xc7k410t
Part xc7k410tiffv900-2L has an unsupported device xc7k410ti
Part xc7k410tiffv676-2L has an unsupported device xc7k410ti
Part xc7k410tiffg900-2L has an unsupported device xc7k410ti
Part xc7k410tiffg676-2L has an unsupported device xc7k410ti
Part xc7k410tifbv900-2L has an unsupported device xc7k410ti
Part xc7k410tifbv676-2L has an unsupported device xc7k410ti
Part xc7k410tifbg900-2L has an unsupported device xc7k410ti
Part xc7k410tifbg676-2L has an unsupported device xc7k410ti
Part xc7k420tffg1156-1 has an unsupported device xc7k420t
Part xc7k420tffg1156-2 has an unsupported device xc7k420t
Part xc7k420tffg1156-2L has an unsupported device xc7k420t
Part xc7k420tffg1156-3 has an unsupported device xc7k420t
Part xc7k420tffg901-1 has an unsupported device xc7k420t
Part xc7k420tffg901-2 has an unsupported device xc7k420t
Part xc7k420tffg901-2L has an unsupported device xc7k420t
Part xc7k420tffg901-3 has an unsupported device xc7k420t
Part xc7k420tffv1156-1 has an unsupported device xc7k420t
Part xc7k420tffv1156-2 has an unsupported device xc7k420t
Part xc7k420tffv1156-2L has an unsupported device xc7k420t
Part xc7k420tffv1156-3 has an unsupported device xc7k420t
Part xc7k420tffv901-1 has an unsupported device xc7k420t
Part xc7k420tffv901-2 has an unsupported device xc7k420t
Part xc7k420tffv901-2L has an unsupported device xc7k420t
Part xc7k420tffv901-3 has an unsupported device xc7k420t
Part xc7k420tiffv901-2L has an unsupported device xc7k420ti
Part xc7k420tiffv1156-2L has an unsupported device xc7k420ti
Part xc7k420tiffg901-2L has an unsupported device xc7k420ti
Part xc7k420tiffg1156-2L has an unsupported device xc7k420ti
Part xc7k480tiffv901-2L has an unsupported device xc7k480ti
Part xc7k480tiffv1156-2L has an unsupported device xc7k480ti
Part xc7k480tiffg901-2L has an unsupported device xc7k480ti
Part xc7k480tiffg1156-2L has an unsupported device xc7k480ti
Part xc7k480tffv901-1 has an unsupported device xc7k480t
Part xc7k480tffv901-2 has an unsupported device xc7k480t
Part xc7k480tffv901-2L has an unsupported device xc7k480t
Part xc7k480tffv901-3 has an unsupported device xc7k480t
Part xc7k480tffv1156-1 has an unsupported device xc7k480t
Part xc7k480tffv1156-2 has an unsupported device xc7k480t
Part xc7k480tffv1156-2L has an unsupported device xc7k480t
Part xc7k480tffv1156-3 has an unsupported device xc7k480t
Part xc7k480tffg901-1 has an unsupported device xc7k480t
Part xc7k480tffg901-2 has an unsupported device xc7k480t
Part xc7k480tffg901-2L has an unsupported device xc7k480t
Part xc7k480tffg901-3 has an unsupported device xc7k480t
Part xc7k480tffg1156-1 has an unsupported device xc7k480t
Part xc7k480tffg1156-2 has an unsupported device xc7k480t
Part xc7k480tffg1156-2L has an unsupported device xc7k480t
Part xc7k480tffg1156-3 has an unsupported device xc7k480t
Find pins for xc7k70tfbg484-1
Find pins for xc7k70tfbg676-1
Find pins for xc7k70tfbv484-1
Find pins for xc7k70tfbv676-1
----------------------------------------
```

File: .github/kokoro/steps/xilinx.sh [hunk 1]
```diff
@@ -7,9 +7,99 @@
...
 export XILINX_LOCAL_USER_DATA=no
+echo "----------------------------------------"
+
+echo
+echo "Select Xilinx Vivado version"
+echo "----------------------------------------"
+(
+	set -e
+	cd /opt
+	if [ x"$XRAY_SETTINGS" = x"kintex7" ]; then
```
Comment (acomodi) [inline .github/kokoro/steps/xilinx.sh:30]: We might add a new `env_vars` in the kokoro config to select which configuration uses which Vivado version?

**Review by acomodi — APPROVED** [review]
LGTM, with a small comment.

I am unsure how complex this might get in the scope of switching CI to GH custom runners, if this is still planned for prjxray, but I guess this is something to keep in mind.

File: .github/kokoro/steps/xilinx.sh (hunk 1 above)
Comment (mithro) [reply .github/kokoro/steps/xilinx.sh:30]: I guess we could in a future version.

Comment (mithro) [reply .github/kokoro/steps/xilinx.sh:30]: The same solution should hopefully work for custom GitHub Action runners...

**Pull request merged**